### ISSUE TITLE
SSR Support

### ISSUE THREAD

**Issue by alexbudure** [AlexandreBonaventure/vue-mq#2]
Can't use the plugin in an SSR environment since the 'window' is undefined on the server. Any ideas on how to properly implement this?

Maybe the $isServer property can help here

**Comment by alexbudure** [comment]
Partially fixed by initializing it just on the client, however it would be nice to be able to pass the media query from the server so it loads properly before getting to the client.

For example, on the server, I know what device the user is on. With that information I can assume a width (if mobile < 600px) and then maybe there's an option for the plugin where you can bypass the whole media query listeners and give it just that width. After it hits the client, it starts the media query listeners and fixes itself if the initial guess was wrong. Thoughts?

**Comment by AlexandreBonaventure** [comment]
Thanks for the suggestion; it should definetly work with SSR. It will be implemented in the next major version.

**Comment by mrleblanc101** [comment]
+1

**Comment by vanhoofmaarten** [comment]
I think that on top of default settings, Client Hints could be very helpful for this feature. 
https://httpwg.org/http-extensions/client-hints.htm
TL;DR: https://httpwg.org/http-extensions/client-hints.html#examples

Browser support at the moment is only Chrome and Opera, but that's a global ±69.6%
https://caniuse.com/#feat=client-hints-dpr-width-viewport

**Comment by Allanon29** [comment]
+1 Any progress with SSR support?

**Comment by giovanecosta** [comment]
+1 for setting a default value for `$md` when no `window` available (server side)

**Comment by paolodina** [comment]
I'm not sure if this could be relevant, but I found on github a  "vue-mq module for Nuxt.js".

https://github.com/vanhoofmaarten/nuxt-mq

I had the 'window' is undefined error too, got resolved by this module.

**Comment by paolodina** [comment]
Sorry, I realized right now that [ssr is set to false](https://github.com/vanhoofmaarten/nuxt-mq/blob/master/lib/module.js#L10).

**Comment by giovanecosta** [comment]
@paolodina Yes, this is the point. If we could have `$mq` default value that is used in case of `window` is missing, i think it's possible to remove `ssr: false` from [nuxt-mq](https://github.com/vanhoofmaarten/nuxt-mq) and start to build apps that is served exactly for the device requested. Such App will also be responsive after that. It's the best case possible IMHO.

**Comment by mystrdat** [comment]
Consider returning `true` for the smallest defined breakpoint when `window` isn't available, that way mobile clients where data and rendering power matters the most get server rendered out of the box and there's no need to work around SSR compatibility otherwise.

**Comment by stefanheimann** [comment]
#27 Should be fixed with this Pull Request - if there's no window object defined, the smallest breakpoint will be considered as default

**Comment by leonelmr** [comment]
Any plans about merging @stefanheimann Pull Request for SSR support #27?

**Comment by AlexandreBonaventure** [comment]
SSR support as of v1.0.0

**Comment by AlexandreBonaventure** [comment]
Thanks for your patience!

**Comment by giovanecosta** [comment]
Thnak you so much @AlexandreBonaventure !!! :tada: :tada: :tada:
